hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -292,6 +292,10 @@ private HoodieWriteConfig createMetadataWriteConfig(
             .withInlineCompaction(false)
             .withMaxNumDeltaCommitsBeforeCompaction(writeConfig.getMetadataCompactDeltaCommitMax())
             .withEnableOptimizedLogBlocksScan(String.valueOf(writeConfig.enableOptimizedLogBlocksScan()))
+            // Compaction on metadata table is used as a barrier for archiving on main dataset and for validating the
+            // deltacommits having corresponding completed commits. Therefore, we need to compact all fileslices of all
+            // partitions together requiring UnBoundedCompactionStrategy.
+            .withCompactionStrategy(new UnBoundedCompactionStrategy())
```

Comment:
Okay, by default we have a 500GB target IO upper bound. Can we add a UT for the write config test of MDT then?

Reply:
500GB may be enough for most indexes we have today. But record index is usually in TBs and hence this issue was seen there.

Reply:
What kind of UT are you suggesting?

Reply:
A write config test, just in case these options are mis-changed during code evolving. Never mind, we can address that in separate PR, can you fire a PR to add a test for MDT write config validation for all these changes you raised recently?